src/api.c

```diff
@@ -2714,6 +2714,46 @@ static int cgroup_get_parent_name(struct cgroup *cgroup, char **parent)
 	return ret;
 }
 
+/*
+ * Checks if the cgroup's controller shares the mount point with any other
+ * controller of the cgroup. Returns 1 if shared or 0.
+ */
+static int is_cgrp_ctrl_mnt_shared(struct cgroup *cgroup, char *controller)
+{
+	char *controller_mnt_point = NULL;
+	int ret = 0;
+	int i, j;
+
+	pthread_rwlock_rdlock(&cg_mount_table_lock);
+
+	for (i = 0; cg_mount_table[i].name[0] != '\0'; i++) {
+		if (strncmp(cg_mount_table[i].name, controller,
+			    sizeof(cg_mount_table[i].name)) == 0) {
+
+			controller_mnt_point =
+				strdup(cg_mount_table[i].mount.path);
+			if (controller_mnt_point == NULL)
+				goto error;
+
+			break;
+		}
+	}
+
+	for (j = 0; controller_mnt_point && j < i; j++) {
```

Comment:
Thinking out loud here.... what if the shared mount point is later in `cg_mount_table` than `i`?  Will this code still find it?

Reply:
This function gets called when the `cgroup` is missing.  If the shared mount point is the after `i` and this is the first of the controllers sharing the mount point, we need to worry because something went wrong.